### PR TITLE
Remove template whitespace

### PR DESCRIPTION
**Pull request by web-bert**
In my mind I always think it's not good to start an HTML document with whitespace
I think a brower in the past didn't like it, it's something that I have always done
This uses the nunjucks whitespace controls to strip the leading whitespace
And also some other whitespace in the header

### CODE REVIEW

**Comment by nickcolley** [comment]
Hey Graham :)

Doing a quick Google confirms that some older browsers will be forced into 'quirks mode' if there is whitespace before the doctype.

Although I would have thought we would have noticed the impact of this in our cross browser testing...

If the issues are around whitespace at the beginning of the document could we change this pull request so it only removes it before the doctype declaration?

I'd like to add a test since this could be easily re-introduced in the future, but we could do this after your pull request lands if you're not comfortable doing this.

**Comment by web-bert** [comment]
Hey @nickcolley 

I have just pushed an update with a test file. I learnt about the `trimBlocks` and `lstripBlocks` config options which I haven't used before, so I've added a test to cover both scenarios as they alter the output - although technically the default one will cover both scenarios, I think it's good to highlight there can be differences?

**Comment by nickcolley** [comment]
Ah so this is only a problem if you do not use those Nunjucks configurations? 

I wonder if we should also look to improve the guidance around those too? (Not in this Pull Request)

File: src/template.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    })
+  })
+  describe('with nunucks block trimming enabled', () => {
+    it('should not have any whitespace before the doctype', () => {
+      nunjucks.configure(configPaths.src, {
+        trimBlocks: true,
+        lstripBlocks: true
+      })
+      const output = nunjucks.render('./template.njk')
+      expect(output.charAt(0)).toEqual('<')
```
Comment (nickcolley) [inline src/template.test.js]: could we do `output.startsWith('<!DOCTYPE html>')`?

**Comment by web-bert** [comment]
It does affect both, just you get more whitespace with one than the other. When using the block trimming the macro import statements don't need the `-%}` at the end, but without they do. Both require the `set` to have the trimming at the start and end.

It would be good to add guidance around the use of the trim config.

File: src/template.njk [hunk 2]
```diff
@@ -18,7 +18,7 @@
       <link rel="apple-touch-icon" sizes="167x167" href="{{ assetPath | default('/assets') }}/images/govuk-apple-touch-icon-167x167.png">
       <link rel="apple-touch-icon" sizes="152x152" href="{{ assetPath | default('/assets') }}/images/govuk-apple-touch-icon-152x152.png">
       <link rel="apple-touch-icon" href="{{ assetPath | default('/assets') }}/images/govuk-apple-touch-icon.png">
-    {% endblock %}
+    {% endblock -%}
```
Comment (nickcolley) [inline src/template.njk]: Is this required to pass the tests?

File: src/template.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+describe('Template', () => {
+  describe('with default nunjucks configuration', () => {
+    it('should not have any whitespace before the doctype', () => {
+      nunjucks.configure(configPaths.src)
+      const output = nunjucks.render('./template.njk')
+      expect(output.charAt(0)).toEqual('<')
+    })
+  })
+  describe('with nunucks block trimming enabled', () => {
```
Comment (nickcolley) [inline src/template.test.js]: nunucks -> nunjucks

File: src/template.test.js (hunk 1 above)
Comment (web-bert) [reply src/template.test.js]: I did wonder whether to check for more, I just thought that as long as it stars with an angle bracket then we know it has no leading whitespace. I didn't want to tie the test to a specific doctype as we don't really care here?

File: src/template.njk (hunk 2 above)
Comment (web-bert) [reply src/template.njk]: Sorry missed that one when reverting...

File: src/template.test.js (hunk 3 above)
Comment (web-bert) [reply src/template.test.js]: oops

**Comment by web-bert** [comment]
Update pushed to fix last two issues

File: src/template.test.js (hunk 1 above)
Comment (nickcolley) [reply src/template.test.js]: Good enough for me 👍

**Comment by nickcolley** [comment]
I've raised an issue for the guidance to investigate separately.

**Comment by nickcolley** [comment]
@web-bert could you please add a CHANGELOG fix entry?

**Comment by web-bert** [comment]
Sure, and then rebase to tidy up as well?

**Comment by nickcolley** [comment]
Yeah, I think we'd be ready to merge then, pending a final review.

**Comment by web-bert** [comment]
No problem, happy to help

File: src/template.test.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/* eslint-env jest */
+
+const nunjucks = require('nunjucks')
+const configPaths = require('../config/paths.json')
+
+describe('Template', () => {
```
Comment (36degrees) [inline src/template.test.js]: 👏

**Pull request merged**